components/keys/src/types.rs

```diff
@@ -114,7 +120,7 @@ impl Key {
             // functions to convert between `TimestampedKey` and `Key`.
             // `TimestampedKey` is in a higher (MVCC) layer, while `Key` is
             // in the core storage engine layer.
-            Err(codec::Error::KeyLength)
+            Err(codec::ErrorInner::KeyLength.into())
```

Comment:
Consider introducing a new `Error` for `Key` so that you can customize what it needs.